### PR TITLE
Add rxjs-no-exposed-subjects rule + tests

### PR DESCRIPTION
**Pull request by maggie-x**
This rule enforces that all subjects which are class properties must be `private`. The test should cover all cases but let me know if there is anything I missed.

### CODE REVIEW

**Comment by cartant** [comment]
Thanks, I'll review it this evening.

**Comment by maggie-x** [comment]
I'd like to attempt these myself! I'll keep you updated on my progress. 

Don't have a twitter handle yet but I can make one aha :) 

Thanks for your feedback! :)

File: test/v6/fixtures/no-exposed-subjects/default/fixture.ts.lint [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  public property$ = new Rx.Subject<void>();
+         ~~~~~~~~~            [Subject 'property$' must be private.]
+
+  protected property2$ = new Rx.Subject<void>();
+            ~~~~~~~~~~           [Subject 'property2$' must be private.]
+
+  anyPropertySubject$ = new Rx.Subject<void>();
+  ~~~~~~~~~~~~~~~~~~~         [Subject 'anyPropertySubject$' must be private.]
+
+  readonly public rPropertySubject$ = new Rx.Subject<void>();
```
Comment (cartant) [inline test/v6/fixtures/no-exposed-subjects/default/fixture.ts.lint]: My understanding is that this is invalid syntax: the `readonly` modifier must be placed *after* the access modifier.

**Comment by maggie-x** [comment]
> * The options don't need to be exposed in the metadata.
> * Some of the options and configuration that came from the Finnish-notation rule are not used/needed.

Do we want users to be able to specify exceptions? Which options and configurations are not needed in your opinion?

**Comment by cartant** [comment]
I don't think any of them need to be exposed. I don't think it's worthwhile differentiating subjects that are exposed via properties, constructor parameters, or accessors - they all have the same effect.

And I don't see the point in treating methods that return subjects differently. I think it's entirely reasonable that this rule is all-or-nothing. And I don't see the point in having name or type-based exceptions. A subject is a subject. And, IMO, subclassing `Subject` in user land is something that users should definitely not be doing, so I don't see any benefit in type-based exceptions.

The Finnish notation rule - upon which this was based - is different, as there are some names that cannot use Finnish notation: names that are imposed by the framework, etc.

**Comment by aboyton** [comment]
I'm not sure a function which is returning a subject needs to be called `fooSubject$()` but I don't hugely mind.

**Comment by maggie-x** [comment]
> * I'd like it to effect failures for non-`private` `get` and `set` accessors, too.

Should these cases have the same error message as the others?

**Comment by cartant** [comment]
> Should these cases have the same error message as the others?

For the `get` and `set` accessors, yes, I think the current failure message is okay.

I think it's okay for methods that return subjects, too, TBH, but if you want a more method-specific failure message that says that non-`private` methods shouldn't return subjects, that's fine, too.

**Comment by cartant** [comment]
Merged. Thanks again for the PR and the work you put into it.

**Pull request merged**